mitodl/mitxpro
# MIT xPRO - 1033 Vouchers: seed data for vouchers

#### Pre-Flight checklist


#### What's this PR do?
fixes #1033 

#### How should this be manually tested?
Make sure you have all the required settings.

1. You need to set these values
`MAILGUN_KEY, MAILGUN_SENDER_DOMAIN, OPENEDX_API_CLIENT_ID, OPENEDX_API_CLIENT_SECRET`
2. You need to all the required voucher setting for both domestic and international test voucher in `.env` file

> VOUCHER_DOMESTIC_DATE_KEY=UNIQUE01
VOUCHER_DOMESTIC_EMPLOYEE_KEY=UNIQUE02
VOUCHER_DOMESTIC_EMPLOYEE_ID_KEY=UNIQUE03
VOUCHER_DOMESTIC_KEY=UNIQUE04
VOUCHER_DOMESTIC_COURSE_KEY=UNIQUE05
VOUCHER_DOMESTIC_CREDITS_KEY=UNIQUE06
VOUCHER_DOMESTIC_DATES_KEY=UNIQUE07
VOUCHER_DOMESTIC_AMOUNT_KEY=UNIQUE08
VOUCHER_INTERNATIONAL_EMPLOYEE_KEY=UNIQUE09
VOUCHER_INTERNATIONAL_PROGRAM_KEY=UNIQUE10
VOUCHER_INTERNATIONAL_COURSE_KEY=UNIQUE11
VOUCHER_INTERNATIONAL_SCHOOL_KEY=UNIQUE12
VOUCHER_INTERNATIONAL_EMPLOYEE_ID_KEY=UNIQUE13
VOUCHER_INTERNATIONAL_AMOUNT_KEY=UNIQUE14
VOUCHER_INTERNATIONAL_DATES_KEY=UNIQUE15
VOUCHER_INTERNATIONAL_COURSE_NAME_KEY=UNIQUE16
VOUCHER_INTERNATIONAL_COURSE_NUMBER_KEY=UNIQUE17

3. Now up your containers and run migrations

4. Now run the seed command `docker-compose run --rm web ./manage.py seed_data`

5. Try to Upload on `xpro.odl.local:port/boeing/upload/`, it will list down the two courses runs for the associated voucher. You need to use domestic and international vouchers.

6. You will see the domestic and international vouchers by uploading these files respectively as shown in the screenshots.

> mitxpro/voucher/.test/domestic_voucher.pdf
mitxpro/voucher/.test/international_voucher.pdf  


#### Screenshots

**Errors**

![error1](https://user-images.githubusercontent.com/4043989/68590747-1a964680-04b1-11ea-92f6-ae82bfc7033e.png)

![Screenshot from 2019-11-11 20-21-04](https://user-images.githubusercontent.com/4043989/68598554-dc088800-04c0-11ea-9ef1-2b9ee812b464.png)


**Domestic Voucher**

![Screenshot from 2019-11-11 20-30-41](https://user-images.githubusercontent.com/4043989/68599521-846b1c00-04c2-11ea-9913-ce582756a1a0.png)


**International Voucher**

![Screenshot from 2019-11-11 20-31-00](https://user-images.githubusercontent.com/4043989/68599549-9056de00-04c2-11ea-8b3a-9eb00c409273.png)
